launch/model_endpoint.py

```diff
@@ -216,8 +218,30 @@ def get(self) -> EndpointResponse:
 class Endpoint:
     """An abstract class that represent any kind of endpoints in Scale Launch"""
 
-    def __init__(self, model_endpoint: ModelEndpoint):
+    def __init__(self, model_endpoint: ModelEndpoint, client):
         self.model_endpoint = model_endpoint
+        self.client = client
+
+    def _update_model_endpoint(self):
+        resp = self.client.connection.get(
+            os.path.join(ENDPOINT_PATH, self.model_endpoint.name)
+        )
+        self.model_endpoint = ModelEndpoint.from_dict(resp)
+
+    def status(self):
+        """Gets the status of the Endpoint."""
+        self._update_model_endpoint()
+        return self.model_endpoint.status
+
+    def resource_settings(self):
+        """Gets the resource settings of the Endpoint."""
+        self._update_model_endpoint()
+        return self.model_endpoint.resource_settings
+
+    def worker_settings(self):
```

Comment:
Tangential note: once we move to pydantic, we'll have a more structured way of returning types from helper functions like these.